vars/buildPlugin.groovy

```diff
@@ -121,7 +123,28 @@ def call(Map params = [:]) {
                   }
                   mavenOptions += 'clean install'
                   try {
-                    infra.runMaven(mavenOptions, jdk, null, null, addToolEnv)
+                    if (artifactCachingProxyEnabled) {
+                      // As azure VM agents don't have this env var, setting a default provider if none is specified or if the provider isn't available
+                      final String defaultProxyProvider = 'azure'
+                      def availableProxyProviders = ['azure', 'do', 'aws']
+                      String requestedProvider = env.ARTIFACT_CACHING_PROXY_PROVIDER
+                      if (requestedProvider == null || requestedProvider == '' || !availableProxyProviders.contains(requestedProvider)) {
+                        requestedProvider = defaultProxyProvider
+                        echo "INFO: no valid artifact caching proxy provider specified, set to '$defaultProxyProvider' by default."
+                      } else {
+                        echo "INFO: using artifact caching proxy from '${requestedProvider}' provider."
+                      }
+
+                      configFileProvider(
+                          [configFile(fileId: "artifact-caching-proxy-${requestedProvider}", variable: 'MAVEN_SETTINGS')]) {
```

Comment:
These files and the associated credentials to use are defined here: https://github.com/jenkins-infra/kubernetes-management/blob/3cf7d326badfce9deb6946e7cfc2956287b833d1/config/ext_jenkins-infra.yaml#L383-L464

Reply:
I.e., https://github.com/jenkins-infra/kubernetes-management/pull/3078. Makes sense to define those things in infra rather than this library I guess.